src/cache_cf.cc

```diff
@@ -959,18 +959,18 @@ configDoConfigure(void)
 
     if (Security::ProxyOutgoingConfig.encryptTransport) {
         debugs(3, 2, "initializing https:// proxy context");
-        Config.ssl_client.sslContext = Security::ProxyOutgoingConfig.createClientContext(false);
-        if (!Config.ssl_client.sslContext) {
+        Config.ssl_client.sslContext = new Security::ContextPointer(Security::ProxyOutgoingConfig.createClientContext(false));
+        if (!Config.ssl_client.sslContext->get()) {
```

Comment:
In createClientContext() failure context, it feels wrong to replace a nil pointer with a non-nil pointer (to a nil pointer). I think we should refactor this a bit to keep Config.ssl_client.sslContext nil if createClientContext() fails and uphold "either Config.ssl_client.sslContext is nil or both pointers are not nil" invariant. Something along these lines may work:

```suggestion
        const auto rawPointer = Security::ProxyOutgoingConfig.createClientContext(false);
        Config.ssl_client.sslContext = rawPointer ? new Security::ContextPointer(rawPointer) : nullptr;
        if (!Config.ssl_client.sslContext) {
```

Reply:
Done.